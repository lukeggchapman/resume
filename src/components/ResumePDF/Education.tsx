import React from 'react'

import ContentItem from './ContentItem'
import macquarieUniLogo from '../../images/macquarie-uni.png'

const Experience = () => (
  <ContentItem
    title="Macquarie University"
    subTitle="Bachelor of Information Technology"
    logo={macquarieUniLogo}
    startDate="2004-01-01"
    endDate="2007-12-31"
  />
)

export default Experience
