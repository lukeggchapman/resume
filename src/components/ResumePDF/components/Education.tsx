import React from 'react'

import hastToPDF from 'utils/hastToPDF'
import ContentItem from './common/ContentItem'
import { EducationData } from '../resumeQuery'

export interface EducationProps {
  data: EducationData
}

const Experience = ({ data }: EducationProps) => {
  return (
    <>
      {data.edges.map(({ node }) => {
        const {
          school,
          degree,
          logo: { publicURL: logoURL },
          startDate,
          endDate,
        } = node.frontmatter

        return (
          <ContentItem
            key={school}
            title={school}
            subTitle={degree}
            logo={logoURL}
            startDate={startDate}
            endDate={endDate}
          >
            {hastToPDF(node.htmlAst)}
          </ContentItem>
        )
      })}
    </>
  )
}

export default Experience
