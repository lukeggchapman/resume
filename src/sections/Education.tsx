import React from 'react'

import hastToPDF from 'utils/hastToPDF'
import ContentItem from '../components/ContentItem'

const Experience = ({ data }: any) => {
  return (
    <>
      {data.edges.map(({ node }: any) => {
        const {
          school,
          degree,
          logo: { publicURL: logoURL },
          startDate,
          endDate,
        } = node.frontmatter

        return (
          <ContentItem
            key={school}
            title={school}
            subTitle={degree}
            logo={logoURL}
            startDate={startDate}
            endDate={endDate}
          >
            {hastToPDF(node.htmlAst)}
          </ContentItem>
        )
      })}
    </>
  )
}

export default Experience
