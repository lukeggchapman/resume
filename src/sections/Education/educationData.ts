import parseMarkdownToHtmlAst from 'utils/parseMarkdownToHtmlAst'

import macquarieData from 'content/education/2015-macquarie.md'
import macquarieLogo from 'content/education/macquarie-uni.png'

interface ExperienceFrontmatter {
  school: string
  degree: string
  startDate: string
  endDate: string
}

const experienceData = [
  {
    ...parseMarkdownToHtmlAst<ExperienceFrontmatter>(macquarieData),
    logo: macquarieLogo,
  },
]

export default experienceData
