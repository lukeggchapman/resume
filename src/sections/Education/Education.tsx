import React from 'react'

import hastToPDF from 'utils/hastToPDF'
import ContentItem from 'components/ContentItem'
import Section from 'components/Section'

import educationData from './educationData'

function Education() {
  return (
    <Section title="Education">
      {educationData.map(({ frontmatter, logo, htmlAst }) => {
        const { school, degree, startDate, endDate } = frontmatter

        return (
          <ContentItem
            key={school}
            title={school}
            subTitle={degree}
            logo={logo}
            startDate={startDate}
            endDate={endDate}
          >
            {hastToPDF(htmlAst)}
          </ContentItem>
        )
      })}
    </Section>
  )
}

export default Education
