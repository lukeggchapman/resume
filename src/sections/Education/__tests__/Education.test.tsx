import React from 'react'
import { shallow } from 'enzyme'

import Education from '../Education'

jest.mock('components/ContentItem', () => 'ContentItem')
jest.mock('utils/hastToPDF', () => jest.fn((htmlAst) => htmlAst))
jest.mock('../educationData', () => [
  {
    frontmatter: {
      school: 'schoolTest1',
      endDate: 'endDateTest1',
      degree: 'degreeTest1',
      startDate: 'startDateTest1',
    },
    logo: 'logoPublcUrlTest1',
    htmlAst: 'htmlAstTest1' as any,
  },
])

describe('Education', () => {
  const render = () => shallow(<Education />)

  it('renders correctly', () => {
    expect(render()).toMatchSnapshot()
  })
})
