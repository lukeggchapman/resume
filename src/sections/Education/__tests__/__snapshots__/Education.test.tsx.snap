// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Education renders correctly 1`] = `
<Section
  title="Education"
>
  <ContentItem
    endDate="endDateTest1"
    key="schoolTest1"
    logo="logoPublcUrlTest1"
    startDate="startDateTest1"
    subTitle="degreeTest1"
    title="schoolTest1"
  >
    htmlAstTest1
  </ContentItem>
</Section>
`;
